core/java/iesi-core/src/main/java/io/metadew/iesi/openapi/OpenAPIGenerator.java

```diff
@@ -0,0 +1,60 @@
+package io.metadew.iesi.openapi;
+
+import com.fasterxml.jackson.core.util.DefaultPrettyPrinter;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import io.metadew.iesi.common.configuration.metadata.repository.MetadataRepositoryConfiguration;
+import io.metadew.iesi.metadata.definition.component.Component;
+import io.metadew.iesi.metadata.definition.connection.Connection;
+import io.metadew.iesi.metadata.operation.MetadataRepositoryOperation;
+import io.metadew.iesi.metadata.repository.MetadataRepository;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class OpenAPIGenerator {
+    private static OpenAPIGenerator INSTANCE;
+
+    private OpenAPIGenerator() {}
+
+
+    public synchronized static OpenAPIGenerator getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new OpenAPIGenerator();
+        }
+        return INSTANCE;
+    }
+
+    public void generate(List<Connection> connections, List<Component> components) {
+        MetadataRepositoryOperation metadataRepositoryOperation = new MetadataRepositoryOperation();
+        List<MetadataRepository> metadataRepositories = new ArrayList<>();
+        metadataRepositories.add(MetadataRepositoryConfiguration.getInstance().getDesignMetadataRepository());
+        metadataRepositories.add(MetadataRepositoryConfiguration.getInstance().getConnectivityMetadataRepository());
```

Comment:
Instead of using the MetadataRepositories to load the configurations from file you can directly use the ConnectionConfiguration and the ComponentConfiguration classes to insert the Connection and Component classes in the repository